### PR TITLE
Adds docs cleaning up the PR walkthrough

### LINKED ISSUE #1852: Update PR checklist to cover actual needs
Author: @pavelpicka (ppicka)

Redmine Issue: 5641, https://pulp.plan.io/issues/5641

---

There is a checklist in our docs \[0\] but doesn't cover towncrier and possibility to update auto-generated API docs.

\[0\] https://docs.pulpproject.org/en/3.0/nightly/contributing/pull-request-walkthrough.html#checklist

### PR DESCRIPTION
**Pull request by bmbouter**
closes #1852

### CODE REVIEW

File: docs/contributing/pull-request-walkthrough.rst [hunk 1]
```diff
@@ -6,57 +6,30 @@ Changes to pulpcore are submitted via `GitHub Pull Requests (PR)
...
-Reviewing a Pull Request
-------------------------
+To Cherry-Pick or Not
+---------------------
 
-When reviewing a PR, it is important to consider where the change ought to land. If you are
-reviewing a bug fix that might be released as part of a z-stream release, you should add the "Needs
-Cherry Pick" label to the PR. Otherwise the label should be removeed. PR authors can also add or
-remove this label if they have write access.
+If you fixing a bug that needs to also be backported to another brancher than ``main``, add the
```
Comment (ggainey) [inline docs/contributing/pull-request-walkthrough.rst:33]:
```suggestion
If you are fixing a bug that should also be backported to another branch than ``main``, add the
```

**Review by ggainey — CHANGES_REQUESTED** [review]
A few suggestions, really glad to see this page reflecting relaity again!

File: docs/contributing/pull-request-walkthrough.rst (hunk 1 above)
Comment (bmbouter) [reply docs/contributing/pull-request-walkthrough.rst]: this fix was pushed, ty

File: docs/contributing/pull-request-walkthrough.rst [hunk 2]
```diff
@@ -6,57 +6,30 @@ Changes to pulpcore are submitted via `GitHub Pull Requests (PR)
...
-Reviewing a Pull Request
-------------------------
+To Cherry-Pick or Not
+---------------------
 
-When reviewing a PR, it is important to consider where the change ought to land. If you are
-reviewing a bug fix that might be released as part of a z-stream release, you should add the "Needs
-Cherry Pick" label to the PR. Otherwise the label should be removeed. PR authors can also add or
-remove this label if they have write access.
+If you are fixing a bug that should also be backported to another branche than ``main``, add the
```
Comment (mdellweg) [inline docs/contributing/pull-request-walkthrough.rst:33]:
```suggestion
If you are fixing a bug that should also be backported to another branch than ``main``, add the
```

File: docs/contributing/pull-request-walkthrough.rst [hunk 3]
```diff
@@ -6,57 +6,30 @@ Changes to pulpcore are submitted via `GitHub Pull Requests (PR)
...
+"Needs Cherry Pick" label to the PR. PR authors can also add or remove this label if they have write
```
Comment (mdellweg) [inline docs/contributing/pull-request-walkthrough.rst:34]: Actually we have `backport-3.20` like labels now.

File: CHANGES/1852.doc [hunk 4]
```diff
@@ -0,0 +1 @@
+Added docs cleaning up the pull request checklist and page.
```
Comment (lubosmj) [inline CHANGES/1852.doc:1]:
```suggestion
Cleared out some of the paragraphs from the Pull Request Walkthrough section.
```

File: docs/contributing/pull-request-walkthrough.rst (hunk 3 above)
Comment (bmbouter) [reply docs/contributing/pull-request-walkthrough.rst:34]: I switched this language.

**Comment by bmbouter** [comment]
@ggainey @mdellweg can you re-review this?

**Review by ggainey — APPROVED** [review]
lg2m!

**Review by mdellweg — APPROVED** [review]

**Pull request merged**